hermeto/core/package_managers/python/uv/main.py

```
@@ -0,0 +1,511 @@
+                    raw_package["sdist"]["url"] = f"file://{_TEMPLATED_DEPS_DIR}/{filename}"
+                raw_wheels = raw_package.get("wheels", [])
+                for raw_wheel, wheel in zip(raw_wheels, package.wheels, strict=True):
+                    filename = wheel.get_target_filename(package.source)
+                    raw_wheel["url"] = f"file://{_TEMPLATED_DEPS_DIR}/{filename}"
+            case "git":
+                _replace_source_with_path(raw_package, _git_tarball_filename(package))
+            case "url":
+                recorded = package.sole_artifact
+                if recorded is None:
+                    # should not happen: artifacts_to_download rejects such a package
+                    raise RuntimeError(f"{package.name} records no sdist and no wheels in uv.lock")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)